### DISCUSSION TITLE
FreeBSD v1.26.0 build failing

### DISCUSSION THREAD

**Discussion by wgmyers** [umami-software/umami#970] [Q&A] [answered]
Anyone else managed to update to 1.26.0 on FreeBSD?

I'm a couple of hours into trying to make this work, beginning to flail, and considering rolling back to 1.24.1, which was fine.

After various false starts I've finally updated the locally built copy of prisma-engines to 3.9.0 (there is no 3.9.2, and as any FreeBSD Umami user already knows, npm flavour prisma does not support FreeBSD) but now, even with my own prisma-engines, when I come to run `npm run build` I get the following error:

```
> dotenv prisma generate

Warning Precompiled binaries are not available for freebsd12. Read more about building your own binaries at https://pris.ly/d/build-binaries
Error: Node API is currently not supported on your platform: freebsd12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! umami@1.26.0 build-db-client: `dotenv prisma generate`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the umami@1.26.0 build-db-client script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Looks like it's no longer enough to build prisma-engines but you have to build the whole of prisma too?

yay

**Comment by mikecao** [comment]
@wgmyers What you can do is rollback prisma itself to the previous version `3.6.0`. It should still work. Just run

```
npm install prisma@3.6.0 @prisma/client@3.6.0 --save
```

☝️ this pins the version to 3.6.0

**Comment by wgmyers** [comment, marked as answer]
Thanks, MIke.

After three hours tearing my hair out I just found the answer. My bad for thinking, oh I'll just quickly run this upgrade at midnight. Bad idea.

Turns out that Prisma have changed their required environment variables for custom installs, and now you need to specify whether you are using libraries or binaries via two new env vars. Also, for some reason, something somewhere (dotenv? prisma?) no longer likes it when you have more than one `.env` file floating around and complains. I had two, one for umami and one for prisma in the prisma directory. The latter was no longer being read, hence Prisma swearing up and down that it had no binaries and didn't support FreeBSD. (It's fine on FreeBSD so long as you compile it yourself. You might need to update rust and cargo again first. I did.)

Anyway. The solution is to add something like the following in `.env` file in the root of the umami repo:

```
PRISMA_CLI_QUERY_ENGINE_TYPE=binary
PRISMA_CLIENT_ENGINE_TYPE=binary
PRISMA_QUERY_ENGINE_BINARY=[$YOURPATHTO]/prisma/query-engine
PRISMA_MIGRATION_ENGINE_BINARY=[$YOURPATHTO]/prisma/migration-engine
PRISMA_INTROSPECTION_ENGINE_BINARY=[$YOURPATHTO]/prisma/introspection-engine
PRISMA_FMT_BINARY=[$YOURPATHTO]/prisma/prisma-fmt
```

Note the first two - these are new since I last had to do this.

Now prisma knows what kind of files it is looking for and can also find them.
